crates/common/src/lib.rs

```diff
@@ -504,57 +502,41 @@ impl std::fmt::Display for Chain {
     }
 }
 
-// these types are used in sequencer tests, which require special fixed width representation
-// FIXME: it'd be better if these had normal varlen display and lenient parsing.
-macros::fmt::thin_debug!(ContractAddress);
-macros::fmt::thin_display!(ContractAddress);
-macros::starkhash::to_from_sql!(ContractAddress);
-
-macros::fmt::thin_debug!(StarknetTransactionHash);
-macros::fmt::thin_display!(StarknetTransactionHash);
-macros::starkhash::to_from_sql!(StarknetTransactionHash);
-
-macros::fmt::thin_debug!(ClassHash);
-macros::fmt::thin_display!(ClassHash);
-macros::starkhash::to_from_sql!(ClassHash);
-
-macros::fmt::thin_debug!(SierraHash);
-macros::fmt::thin_display!(SierraHash);
-macros::starkhash::to_from_sql!(SierraHash);
-
-macros::fmt::thin_debug!(CasmHash);
-macros::fmt::thin_display!(CasmHash);
-macros::starkhash::to_from_sql!(CasmHash);
-
-macros::fmt::thin_debug!(ClassCommitment);
-macros::fmt::thin_display!(ClassCommitment);
-macros::starkhash::to_from_sql!(ClassCommitment);
+macros::starkhash::common_newtype_with_compressed_sql!(
+    ContractNonce,
+    StorageValue,
+);
```

Comment:
Should I pro-actively include `TransactionNonce`? It is actually not used encoded storage currently at all. Neither is `StorageValue` right now, however it will become used in #1038 which was the driver behind this PR.